chore: Add documentation for the transformServiceBindingToClientCredentialsDestination function (#1949)
SAP/cloud-sdk: docs-js/features/connectivity/destination.mdx

The default implementation also retrieves a service token, if needed.

Additionally, we provide a default implementation for the transformation of service bindings to OAuth2ClientCredentials destinations.

> It assumes that your service binding is of the following structure:

Reviewer: ```suggestion
```

```json
{
  "VCAP_SERVICES": {
    "custom-service": [
      {
        "name": "my-custom-service",
        "label": "custom-service",
        "tags": ["custom-service"],
        "url": "https://example.com",
        "credentials": {
          "clientid": "CLIENT_ID",
          "clientsecret": "CLIENT_SECRET"
        }
      }
    ]
  }
}
```

> If your service URL is not in the `url` property, you can alternatively provide a `url` property as part of the `options` object.

Reviewer: ```suggestion
If the service URL is not specified in the `url` property, it can alternatively be provided as part of the `options` parameter.
```
